src/lib/utils.ts

```diff
@@ -402,25 +402,19 @@ export function getSchemaElement(schemas: Erc725Schema[], key: string) {
   return schemaElement;
 }
 
-/**
- * Hash data depending on chosen hashFunction
- *
- * @export
- * @param {*} data
- * @param {*} hashFunction
- * @return {*}  {string}
- */
-export function hashData(data, hashFunction): string {
+export function hashData(data: unknown, hashFunction: string): string {
   switch (hashFunction.toLowerCase()) {
     case 'keccak256(utf8)':
+    case '0x6f357c6a':
       return keccak256(JSON.stringify(data));
 
     case 'keccak256(bytes)':
+    case '0x8019f9b1':
       return keccak256(data);
 
     default:
       throw new Error(
-        `Chosen hashFunction ${hashFunction.toLowerCase()} is not supported`,
+        `Chosen hashFunction '${hashFunction.toLowerCase()}' is not supported`,
```

Comment:
I would suggest to do:

```ts
enum SUPPORTED_HASH_FUNCTIONS {
  KECCAK256_UTF8 = 'keccak256(utf8)',
  HASH_KECCAK256_UTF8 = '0x6f357c6a',
  KECCAK256_BYTES = 'keccak256(bytes)',
  HASH_KECCAK256_BYTES = '0x8019f9b1',
}

export function hashData(data: unknown, hashFunction: SUPPORTED_HASH_FUNCTIONS)
```

+ use enum in switch

+ in error, add the list of supported hash functions:

```
`hashFunction '${hashFunction.toLowerCase()}' is not supported. Supported hashFunction: ${listValuesOfEnum(SUPPORTED_HASH_FUNCTIONS)}`
```

May be "overkill" but I think it adds clarity and makes things cleaner.

Here, hashFunctions is a finite set of strings, so it would make more sense for me to us an enum rather than more global "string" ?

The values from this enum can also be used elsewhere in the code, cf. other comment.

Reply:
Good idea, I changed this accordingly